src/main/java/pt/tqs/g205/services/EncomendaRestauranteService.java

```diff
@@ -46,14 +50,13 @@
     
     Optional<EstadoEncomenda> optEstado = estadoEncomendaRepo.findById(1);
     
-    if (!optEstado.isPresent()) {
-      throw new NoSuchElementException();
-    }
-    
     EstadoEncomenda estado = optEstado.get();
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Call "optEstado.isPresent()" before accessing the value. ([squid:S3655](https://sonarcloud.io/organizations/bdomingues-github/rules#rule_key=squid%3AS3655))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=pt.tqs.g205%3Arestapi&pullRequest=25&issues=AWPNQ-Bx4fWm9CZEZoxU&open=AWPNQ-Bx4fWm9CZEZoxU)

[ISSUE_KEY]: AWPNQ-Bx4fWm9CZEZoxU